src/st/no0/e_flea_man.c

```diff
@@ -0,0 +1,179 @@
+            g_CurrentEntity->posY.i.hi += collider.unk20;
+            break;
+        }
+    }
+}
+
+void SetFacingLeft(void) {
+    g_CurrentEntity->facingLeft = (GetSideToPlayer() & 1) ^ 1;
+}
+
+void EntityFleaMan(Entity* self) {
+    u32 distanceX;
+    u8 rand;
+    u8 index;
+
+    if (self->flags & FLAG_DEAD) {
+        PlaySfxPositional(SFX_SMALL_FLAME_IGNITE);
+        EntityExplosionSpawn(1, 0);
+        return;
+    }
+
+    if (self->ext.fleaMan.unk7C) {
+        self->ext.fleaMan.unk7C--;
+        self->hitboxState = 0;
+    } else {
+        self->hitboxState = 19;
+    }
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitFleaMan);
+        self->animCurFrame = 16;
+        if (self->params & 1) {
+            self->facingLeft = 1;
+            self->velocityX = 0x140;
+            self->velocityY = 0xA0;
+            self->step = 2;
+        } else {
+            self->step = 1;
+        }
+
+#ifndef VERSION_PSP
+        if (self->ext.fleaMan.unk7C) {
+            self->step = 3;
+        }
+#endif
+        break;
+
+    case 1:
+        AnimateEntity(anim_stand, self);
+        SetFacingLeft();
+        if (GetDistanceToPlayerX() < 0x60 && GetDistanceToPlayerY() < 0x40) {
+            self->step = 3;
+        }
+        break;
+
+    case 2:
+        self->zPriority = 0;
+        MoveEntity();
+        break;
+
+    case 3:
+        SetFacingLeft();
+        if (self->hitParams && self->hitParams < 4) {
+            index = 2;
+        } else {
+            rand = Random() & 7;
+            distanceX = GetDistanceToPlayerX();
+
+            if (distanceX < 0x50) {
+                switch (rand) {
+                case 0:
+                case 1:
+                    index = 3;
+                    break;
+                case 2:
+                    index = 2;
+                    break;
+                case 3:
+                    index = 1;
+                    break;
+                default:
+                    index = 0;
+                    break;
+                }
+            } else if (distanceX <= 0x7F) {
+                switch (rand) {
+                case 0:
+                    index = 3;
+                    break;
+                case 1:
+                    index = 2;
+                    break;
+                case 2:
+                    index = 0;
+                    break;
+                default:
+                    if (self->facingLeft ^ PLAYER.facingLeft) {
+                        index = 0;
+                    } else {
+                        index = 1;
+                    }
+                    break;
+                }
+            } else {
+                index = 4;
+            }
+        }
+
+        distanceX = fidgetVelocityX[index];
+        if (self->facingLeft) {
+            self->velocityX = -distanceX;
+        } else {
+            self->velocityX = distanceX;
+        }
+
+        self->velocityY = fidgetVelocityY[index];
```

Comment:
I had thought this might be a single `Point32` array for these x/y but didn't seem to be the case here, both access seem to be at offset 0 so the `Point32.x`/`Point32.y` didn't seem to work